web_api.md

```diff
@@ -0,0 +1,318 @@
+a configuration file. The schema of a configuration file can be found [here](web_api/config.schema.yml). Below is a
+small example of how the file might look like.
+
+```yaml
+message_queue:
+  address: localhost
+  port: 5672
+  credentials:
+    username: admin
+    password: admin
+  ssh:
+    username: cloud
+    path_to_privkey: /path/to/file
+mongo_db:
+  address: localhost
+  port: 27017
+  credentials:
+    username: admin
+    password: admin
+  ssh:
+    username: cloud
+    password: "1234"
+hosts:
+  - address: localhost
+    username: cloud
+    password: "1234"
+    deploy_processors:
+      - name: ocrd-cis-ocropy-binarize
+        number_of_instance: 2
+        deploy_type: native
+      - name: ocrd-olena-binarize
+        number_of_instance: 1
+        deploy_type: docker
+
+  - address: 134.76.1.1
+    username: tdoan
+    path_to_privkey: /path/to/file
+    deploy_processors:
+      - name: ocrd-eynollah-segment
+        number_of_instance: 1
+        deploy_type: native
+```
+
+There are three main sections in the configuration file.
+
+1. `message_queue`: it contains the `address` and `port`, where the queue was deployed, or will be deployed with the
+   specified `credentials`. If the `ssh` property is presented, the Processing Server will try to connect to
+   the `address` via `ssh` with provided `username` and `password` and deploy [RabbitMQ](https://www.rabbitmq.com/) at
+   the specified `port`. The remote machine must have [Docker](https://www.docker.com/) installed since the deployment
+   is done via Docker. Make sure that the provided `username` has enough rights to run Docker commands. In case
+   the `ssh` property is not presented, the Processing Server assumes that RabbitMQ was already deployed and just uses
+   it.
+2. `mongo_db`: this section also contains the `address` and `port`, where the [MongoDB](https://www.mongodb.com/) is
+   running, or will run. If `credentials` is presented, it will be used when deploying and connecting to the database.
+   The `ssh` section behaves exactly the same as described in the `message_queue` section above.
+3. `hosts`: this section contains a list of hosts, usually virtual machines, where Processing Workers should be
+   deployed. To be able to connect to a host, an `address` and `username` are required, then comes either `password`
+   or `path_to_privkey` (path to a private key). All Processing Workers, which will be deployed, must be declared under
+   the `deploy_processors` property. In case `type` is `docker`, make sure that [Docker](https://www.docker.com/) is
+   installed in the target machine and the provided `username` has enough rights to execute Docker commands.
+
+Among three sections, only the `message_queue` is required. However, if `hosts` is present, `mongo_db` must be there
+as well. For more information, please check the [configuration file schema](web_api/config.schema.yml).
+
+### Processing Worker
```

Comment:
I'd recommend moving this section below the Process Queue, because it is easier to explain top-down, and workers only make sense with queues.

Reply:
OK, fine for me.